src/cdk/bidi/dir-document-token.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {APP_ROOT_SCOPE, inject, InjectionToken} from '@angular/core';
+
+
+/**
+ * Injection token used to inject the document into Directionality.
+ * This is used so that the value can be faked in tests.
+ *
+ * We can't use the real document in tests because changing the real `dir` causes geometry-based
+ * tests in Safari to fail.
+ *
+ * We also can't re-provide the DOCUMENT token from platform-brower because the unit tests
+ * themselves use things like `querySelector` in test code.
+ *
+ * This token is defined in a separate file from Directionality as a workaround for
+ * https://github.com/angular/angular/issues/22559
+ *
+ * @docs-private
+ */
+export const DIR_DOCUMENT = new InjectionToken<Document>('cdk-dir-doc', {
```

Comment:
As an aside, do we really need a separate token for the document here? We can easily stub out the `DOCUMENT` itself in the tests.

Reply:
When I was originally doing this I found that re-providing common's `DOCUMENT` with something else made Angular not work.